File: ChiffreDeVigenere/ChiffreDeVigenere.cpp
#include <iostream>
using namespace std;

int main(){  
    string alph = "ABCDEFGHIJKLMNOPQRSTUVWXYZ", arrAlph[26], key, longKey = "", word, result = "";
    int t = 0;
    arrAlph[0] = alph;
    cout << "Please, enter the word" << endl;
    cin >> word;
    cout << "Please, enter the key" << endl;
    cin >> key;

    for(int i = 1; i < 26; ++i){
        string str = arrAlph[i - 1];
        str += str[0];
        str.erase(0, 1);
        arrAlph[i] = str;
    }

    for(int i = 0; i < word.length(); ++i){
        longKey += key[t];
        ++t;
        t = t == key.length() ? 0 : t;
    }

    for(int i = 0; i < word.length(); ++i){
        result += arrAlph[alph.find(longKey[i])][alph.find(word[i])];
    }

    cout << result;
    return 0;
}    

File: ChiffreDeVigenere/ChiffreDeVigenere.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Word:', (word) => {
    rl.setPrompt('Key:');
    rl.prompt();
    rl.on('line', (key) => { 
        let alph = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', arrAlph = [alph], longKey = '', result = '', t = 0;
        
        for(let i = 1; i < 27; ++i){
            let str = arrAlph[i - 1];
            str += str[0];
            str = str.slice(-str.length + 1);
            arrAlph[i] = str;
        }

        for(let i = 0; i < word.length; ++i){
            longKey += key[t];
            ++t;
            t = t == key.length ? 0 : t;
        }

        for(let i = 0; i < word.length; ++i){
            result += arrAlph[alph.indexOf(longKey[i])][alph.indexOf(word[i])];
        }

        console.log(result);

       rl.close();
    })
})
